packages/client/src/views/UserSetup/SetupReviewPage.tsx

```diff
@@ -72,137 +65,139 @@ const LoaderOverlay = styled.div`
   align-items: center;
   flex-direction: column;
 `
-
-interface IState {
-  submitError: boolean
-}
-
 interface IProps {
   setupData: IProtectedAccountSetupData
-  userDetails: IUserDetails | null
   goToStep: (
     step: ProtectedAccoutStep,
     data: IProtectedAccountSetupData
   ) => void
 }
 
-type IFullProps = IProps & IntlShapeProps
-
-class UserSetupReviewComponent extends React.Component<IFullProps, IState> {
-  constructor(props: IFullProps) {
-    super(props)
-    this.state = {
-      submitError: false
-    }
-  }
-
-  render = () => {
-    const { intl, userDetails } = this.props
-
-    const englishName =
-      (userDetails &&
-        userDetails.name &&
-        (createNamesMap(userDetails.name as GQLHumanName[])['en'] as string)) ||
-      ''
-
-    const mobile = (userDetails && (userDetails.mobile as string)) || ''
-
-    const typeRole =
-      (userDetails &&
-        userDetails.role &&
-        (userDetails.type
-          ? `${intl.formatMessage(
-              userMessages[userDetails.role as string]
-            )} / ${intl.formatMessage(
-              userMessages[userDetails.type as string]
-            )}`
-          : `${intl.formatMessage(
-              userMessages[userDetails.role as string]
-            )}`)) ||
-      ''
-
-    const primaryOffice =
-      (userDetails &&
-        userDetails.primaryOffice &&
-        userDetails.primaryOffice.name) ||
-      ''
-
-    const answeredQuestions: IDataProps[] = []
-    this.props.setupData.securityQuestionAnswers &&
-      this.props.setupData.securityQuestionAnswers.forEach((e) => {
-        answeredQuestions.push({
-          id: `Question_${e.questionKey}`,
-          label: intl.formatMessage(userMessages[e.questionKey]),
-          value: e.answer,
-          action: {
-            id: `Question_Action_${e.questionKey}`,
-            label: intl.formatMessage(buttonMessages.change),
-            handler: () =>
-              this.props.goToStep(
-                ProtectedAccoutStep.SECURITY_QUESTION,
-                this.props.setupData
-              )
-          }
-        })
-      })
-    const items = [
-      {
-        id: 'EnglishName',
-        label: intl.formatMessage(messages.labelEnglishName),
-        value: englishName,
-        action: {
-          label: intl.formatMessage(buttonMessages.change),
-          disabled: true
-        }
-      },
-      {
-        id: 'UserPhone',
-        label: intl.formatMessage(constantsMessages.labelPhone),
-        value: mobile,
+export function UserSetupReview({ setupData, goToStep }: IProps) {
+  const intl = useIntl()
+  const [submitError, setSubmitError] = React.useState(false)
+  const userDetails = useSelector<IStoreState, IUserDetails | null>(
+    getUserDetails
+  )
+  const englishName =
+    (userDetails &&
+      userDetails.name &&
+      (createNamesMap(userDetails.name as GQLHumanName[])['en'] as string)) ||
+    ''
```

Comment:
We can use [getUserName](https://github.com/opencrvs/opencrvs-core/blob/f2f287d86c906ab00981fc09e879bc539228f2e9/packages/client/src/utils/userUtils.ts#L171) here